src/core_modules/capture-core/components/D2Form/DataEntryPlugin/DataEntryPlugin.container.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React, { useEffect, useMemo } from 'react';
+import { DataEntryPluginComponent } from './DataEntryPlugin.component';
+import type { ContainerProps } from './DataEntryPlugin.types';
+import { usePluginMessages } from './hooks/usePluginMessages';
+import { usePluginCallbacks } from './hooks/usePluginCallbacks';
+import { usePluginValues } from './hooks/usePluginValues';
+
+export const DataEntryPlugin = (props: ContainerProps) => {
+    const { pluginSource, fieldsMetadata, formId, onUpdateField, pluginContext } = props;
+    const metadataByPluginId = useMemo(() => Object.fromEntries(fieldsMetadata), [fieldsMetadata]);
+    const configuredPluginIds = useMemo(() => Object.keys(metadataByPluginId), [metadataByPluginId]);
+
+    // Plugin related functionality and feedback
+    const { pluginValues } = usePluginValues(formId, metadataByPluginId, pluginContext);
+    const { errors, warnings } = usePluginMessages(formId, metadataByPluginId);
+    const { setFieldValue, setContextFieldValue } = usePluginCallbacks({
+        configuredPluginIds,
+        onUpdateField,
+        metadataByPluginId,
+        pluginContext,
+    });
+
+    // Expanding iframe height temporarily to fit content - LIBS-487
+    useEffect(() => {
+        const iframe = document.querySelector('iframe');
+        if (iframe) iframe.style.height = '500px';
+    }, []);
+
+
+    // Formatted metadata is needed to remove the underscore from the keys
+    const formattedMetadata = useMemo(() => Array.from(fieldsMetadata.entries())
```

Comment:
We also have to convert option sets. Adding an option set right now causes a circular structure error.

Additionally, I think we should not pass on the app-ids, so we don't encourage using those (won't work cross instance)